lib/broccoli/ember-app.js

```diff
@@ -1024,13 +1024,26 @@ class EmberApp {
         overwrite: true,
       });
 
-      if (this.amdModuleNames) {
-        let anonymousAmd = new Funnel(externalTree, {
-          files: Object.keys(this.amdModuleNames),
-          annotation: 'Funnel (named AMD)',
-        });
-        externalTree = this._mergeTrees([externalTree, shimAmd(anonymousAmd, this.amdModuleNames)], {
-          annotation: 'TreeMerger (named AMD)',
+      if (this.entries) {
+        let transformedTrees = [];
+        for (let filePath in this.entries) {
+          let entry = this.entries[filePath];
+          let treeToBeTransformed = new Funnel(externalTree, {
```

Comment:
my concern with this approach, is we will now be creating a funnel per input. I worry this will result in later performance issues.

The existing approach for the `amd` shim, is to do 1 funnel for all `amd` transforms. I suspect we should continue to follow this pattern.